pkg/test/dualstack/cloud_providers_dualstack_test.go

```diff
@@ -493,21 +554,24 @@ func checkNodeReadiness(t *testing.T, userClient *kubernetes.Clientset, expected
 
 func createUsercluster(t *testing.T, apicli *utils.TestClient, projectName string, clusterSpec models.CreateClusterSpec) (*rest.Config, string, string, func(), error) {
 	var teardowns []func() error
+	var once sync.Once
 	cleanup := func() {
-		n := len(teardowns)
-		for i := range teardowns {
-			err := wait.Poll(10*time.Second, 10*time.Minute, func() (bool, error) {
-				err := teardowns[n-1-i]()
+		once.Do(func() {
+			n := len(teardowns)
+			for i := range teardowns {
+				err := wait.Poll(10*time.Second, 10*time.Minute, func() (bool, error) {
+					err := teardowns[n-1-i]()
```

Comment:
What if the `teardown` length is one?

Reply:
In that case 
n = 1
i = 0 
n - 1 - i = 1 - 1 - 0 = 0 

all good

yes?